File: iOS/App/ProductsCrud/Presentation/Product/ViewModel/ProductDetails/ProductDetailsViewModelState.swift
//
//  ProductViewModelState.swift
//  ProductsCrud
//
//  Created by Emil Falkowski on 11/11/2024.
//

struct ProductDetailsViewModelState: Equatable {
    let dataState: DataState
    let productDetails: PresentationProductDetails

    static func initial() -> ProductDetailsViewModelState {
        return ProductDetailsViewModelState(
            dataState: .loading,
            productDetails: PresentationProductDetails.empty())
    }
}

File: Android/App/app/src/main/java/com/efalco777/crud_products/presentation/product/viewmodel/product_details/ProductDetailsViewModelState.kt
package com.efalco777.crud_products.presentation.product.viewmodel.product_details

import com.efalco777.crud_products.presentation.core.viewmodel.DataState
import com.efalco777.crud_products.presentation.product.model.PresentationProductDetails

data class ProductDetailsViewModelState(
    val productDetails: PresentationProductDetails,
    val dataState: DataState
) {
    companion object {
        fun initial() =
            ProductDetailsViewModelState(
                productDetails = PresentationProductDetails.empty(),
                dataState = DataState.Loading
            )
    }
}